### ISSUE TITLE
Errors highlighted by `mypy`

### ISSUE THREAD

**Issue by suvayu** [transientskp/pyse#82]
```
$ hatch run lint:mypy --install-types --non-interactive --pretty 
sourcefinder/utility/containers.py:55: error: Signatures of "__iadd__" and "__add__" are incompatible  [misc]
        def __iadd__(self, y):
        ^
sourcefinder/accessors/lofarhdf5image.py:61: error: "LofarHdf5Image" has no attribute "header"  [attr-defined]
            return self.header
                   ^~~~~~~~~~~
sourcefinder/image.py:1181: error: "ParamSet" has no attribute "sig"  [attr-defined]
                    param.sig = maxis[count] / self.rmsmap.data[tuple(maxposs[count])]
                    ^~~~~~~~~
Found 3 errors in 3 files (checked 30 source files)
```

**Comment by HannoSpreeuw** [comment]
Allright. The latter error is from a snippet of code that converts source measurements as floats in Numpy.ndarrays into `extract.Detection` instances which is the format that our unit tests accept.
As long as we have not adapted the unit tests to accept a new format for the source measurements, this conversion has to stay in place.

But for now I can probably fix it using
```
param["sig"] = maxis[count] / self.rmsmap.data[tuple(maxposs[count])]
```

**Comment by HannoSpreeuw** [comment]
The first and second one I could fix by removing the corresponding methods.

**Comment by HannoSpreeuw** [comment]
Fixed through PR #85
